### PR TITLE
Add branch dropdown in nav v1

### PR DESCRIPTION
**Pull request by joshenlim**

### CODE REVIEW

File: studio/components/interfaces/BranchManagement/BranchManagement.tsx [hunk 1]
```diff
@@ -137,16 +136,16 @@ const BranchManagement = () => {
           </div>
         ) : (
           <div className="flex items-center space-x-2 !mt-4">
-            <Link passHref href={'/'}>
+            <Link passHref href="https://forms.supabase.com/branching-request">
```
Comment (MildTomato) [inline studio/components/interfaces/BranchManagement/BranchManagement.tsx:139]: you know what im going to say...

File: studio/components/interfaces/Home/ProjectUsageSection.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ const ProjectUsageSection = observer(() => {
           <IconLoader className="animate-spin" size={14} />
           <p className="text-sm">Retrieving project usage statistics</p>
         </div>
-      ) : true ? (
+      ) : hasProjectData ? (
```
Comment (MildTomato) [inline studio/components/interfaces/Home/ProjectUsageSection.tsx:45]: sure?

File: studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+      organizationIntegrationId: integration.id,
+    })
+  }
+
+  return (
+    <div className="flex flex-col gap-6 px-8 py-8">
+      <div>
+        <Label_Shadcn_ className="text">Production branch</Label_Shadcn_>
+        <p className="text-xs text-light mb-3">
+          Deploy Edge Functions when merged into Production branch
```
Comment (MildTomato) [inline studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx:135]: This isn't true. It's only for branching right now

File: studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+                    )
+                  })}
+                </CommandGroup_Shadcn_>
+              </CommandList_Shadcn_>
+            </Command_Shadcn_>
+          </PopoverContent_Shadcn_>
+        </Popover_Shadcn_>
+      </div>
+
+      <Form_Shadcn_ {...form}>
```
Comment (MildTomato) [inline studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx:206]: This whole file seems a bit off.

Comment (MildTomato) [reply studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx:206]: Isn't this functionality covered elsewhere ?

Comment (MildTomato) [reply studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx:206]: Also looks like something I did, maybe we moved it here.

File: studio/components/layouts/AppLayout/BranchDropdown.tsx [hunk 5]
```diff
@@ -63,7 +63,11 @@ const BranchLink = ({
   )
 }
 
-const BranchDropdown = () => {
+interface BranchDropdownProps {
+  alt?: boolean
```
Comment (MildTomato) [inline studio/components/layouts/AppLayout/BranchDropdown.tsx:67]: maybe need better prop name

File: studio/components/layouts/AppLayout/EnableBranchingButton/BranchingWaitlistPopover.tsx [hunk 6]
```diff
@@ -9,13 +9,17 @@ import {
   Popover_Shadcn_,
 } from 'ui'
 
-const BranchingWaitlistPopover = () => {
+interface BranchingWaitlistPopoverProps {
+  alt?: boolean
```
Comment (MildTomato) [inline studio/components/layouts/AppLayout/EnableBranchingButton/BranchingWaitlistPopover.tsx:13]: maybe better prop name again

File: studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingButton.tsx [hunk 7]
```diff
@@ -1,187 +1,34 @@
...
-import VercelProjectSelection from './VercelProjectSelection'
+
 import { OPT_IN_TAGS } from 'lib/constants'
 
-const EnableBranchingButton = () => {
-  const { ui } = useStore()
-  const { ref } = useParams()
-  const selectedOrg = useSelectedOrganization()
+interface EnableBranchingButtonProps {
+  alt?: boolean
```
Comment (MildTomato) [inline studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingButton.tsx:10]: still don't know what alt is..

File: studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx (hunk 3 above)
Comment (MildTomato) [reply studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx:135]:
```suggestion
          All other branches will be treated as Preview Branches
```

File: studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+        <form onSubmit={form.handleSubmit(onSubmit)}>
+          <FormLabel_Shadcn_ className="!text">Supabase directory</FormLabel_Shadcn_>
+          <FormDescription_Shadcn_ className="text-xs text-lighter mb-3">
+            Migrations and seed.sql file will be run from this directory.
```
Comment (MildTomato) [inline studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx:210]:
```suggestion
            Migration and seed SQL files will be run from this directory.
```

File: studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingModal.tsx [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+import { useBranchCreateMutation } from 'data/branches/branch-create-mutation'
+import { useOrgIntegrationsQuery } from 'data/integrations/integrations-query-org-only'
+import { IntegrationName } from 'data/integrations/integrations.types'
+import { useSelectedOrganization, useStore } from 'hooks'
+import { useAppUiStateSnapshot } from 'state/app'
+import BranchingWaitlistPopover from './BranchingWaitlistPopover'
+import GithubRepositorySelection from './GithubRepositorySelection'
+import VercelProjectSelection from './VercelProjectSelection'
+
+// [Joshen] Temp
```
Comment (MildTomato) [inline studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingModal.tsx:17]: is it still temp?

File: studio/components/layouts/AppLayout/EnableBranchingButton/GithubRepositorySelection.tsx [hunk 10]
```diff
@@ -52,6 +52,12 @@ const GithubRepositorySelection = ({
   const [repoOwner, repoName] = githubConnection?.metadata.name.split('/') ?? []
 
   const sidePanels = useSidePanelsStateSnapshot()
+  const githubIntegrationAppUrl =
+    process.env.NEXT_PUBLIC_ENVIRONMENT === 'prod'
```
Comment (MildTomato) [inline studio/components/layouts/AppLayout/EnableBranchingButton/GithubRepositorySelection.tsx:56]: There's an actual Vercel env var for 'prod' and 'preview' if anyone can remember it. Better than relying on a custom one.

File: studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx (hunk 4 above)
Comment (joshenlim) [reply studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx:206]: these were just shifted from IntegrationSettings

File: studio/components/interfaces/Home/ProjectUsageSection.tsx (hunk 2 above)
Comment (joshenlim) [reply studio/components/interfaces/Home/ProjectUsageSection.tsx:45]: yeap i double checked with Ziinc - `true` was a bug
would imply that new projects wont see the "Welcome" state

File: studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingModal.tsx (hunk 9 above)
Comment (joshenlim) [reply studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingModal.tsx:17]: nope! will remove

File: studio/components/layouts/AppLayout/EnableBranchingButton/BranchingWaitlistPopover.tsx (hunk 6 above)
Comment (joshenlim) [reply studio/components/layouts/AppLayout/EnableBranchingButton/BranchingWaitlistPopover.tsx:13]: will add a comment

File: studio/components/layouts/AppLayout/BranchDropdown.tsx (hunk 5 above)
Comment (joshenlim) [reply studio/components/layouts/AppLayout/BranchDropdown.tsx:67]: will add a comment

File: studio/components/layouts/AppLayout/EnableBranchingButton/GithubRepositorySelection.tsx (hunk 10 above)
Comment (joshenlim) [reply studio/components/layouts/AppLayout/EnableBranchingButton/GithubRepositorySelection.tsx:56]: i'll probably keep this first and refactor in another PR cause theres multiple places using NEXT_PUBLIC_ENVIRONMENT

ref for myself: https://vercel.com/docs/projects/environment-variables/system-environment-variables

File: studio/components/interfaces/BranchManagement/BranchManagement.tsx [hunk 11]
```diff
@@ -137,9 +137,11 @@ const BranchManagement = () => {
         ) : (
           <div className="flex items-center space-x-2 !mt-4">
             <Link passHref href="https://forms.supabase.com/branching-request">
-              <a rel="noreferrer" target="_blank">
-                <Button>Join waitlist</Button>
-              </a>
+              <Button>
```
Comment (MildTomato) [inline studio/components/interfaces/BranchManagement/BranchManagement.tsx:140]: asChild

File: studio/components/layouts/AppLayout/EnableBranchingButton/BranchingWaitlistPopover.tsx [hunk 12]
```diff
@@ -35,16 +39,16 @@ const BranchingWaitlistPopover = () => {
...
               <Button>Join waitlist</Button>
             </a>
           </Link>
-          <Link passHref href={'/'}>
+          {/* <Link passHref href={'/'}>
             <a rel="noreferrer" target="_blank">
               <Button type="default">View the docs</Button>
             </a>
-          </Link>
+          </Link> */}
```
Comment (alaister) [inline studio/components/layouts/AppLayout/EnableBranchingButton/BranchingWaitlistPopover.tsx:51]: Still needed?

File: studio/components/interfaces/BranchManagement/BranchManagement.tsx [hunk 13]
```diff
@@ -137,16 +136,18 @@ const BranchManagement = () => {
...
+                </a>
+              </Button>
             </Link>
-            <Link passHref href={'/'}>
+            {/* <Link passHref href={'/'}>
               <a rel="noreferrer" target="_blank">
                 <Button type="default">View the docs</Button>
               </a>
-            </Link>
+            </Link> */}
```
Comment (alaister) [inline studio/components/interfaces/BranchManagement/BranchManagement.tsx:150]: Still needed?

**Review by alaister — APPROVED** [review]
LGTM! Couple of non-blocking questions:

I feel like these should be filtered by the org:
Screenshot 2023-09-20 at 17 50 32
wdyt?

**Comment by joshenlim** [comment]
> I feel like these should be filtered by the org:

i was thinking about this - but there'll be a weird UX RE navigation hierarchy imo. If we do this, to access a project in another org, the user would have to go to the home page and find that project. The organization dropdown links the user to the org settings page, rather than linking to a page which has all the projects under that org so the behaviour is slightly odd

i'm thinking this will be solved with the v2 nav as the user can then select a org from the org dropdown, which will then bring them to a page of projects under that org - the context of being under a org is more apparent imo

but i might be thinking too much though - please do let me know 😅

File: studio/components/layouts/AppLayout/EnableBranchingButton/BranchingWaitlistPopover.tsx (hunk 12 above)
Comment (joshenlim) [reply studio/components/layouts/AppLayout/EnableBranchingButton/BranchingWaitlistPopover.tsx:51]: i'm leaving these as placeholders cause we should have docs for branching eventually

File: studio/components/interfaces/BranchManagement/BranchManagement.tsx (hunk 13 above)
Comment (joshenlim) [reply studio/components/interfaces/BranchManagement/BranchManagement.tsx:150]: same as the previous comment

**Comment by MildTomato** [comment]
Agreed. Let's not change how they work for this PR

**Pull request merged**